dd-sdk-android/src/main/kotlin/com/datadog/android/instrumentation/gesture/WindowCallbackWrapper.kt

```diff
@@ -0,0 +1,167 @@
+package com.datadog.android.instrumentation.gesture
+
+import android.os.Build
+import android.view.ActionMode
+import android.view.KeyEvent
+import android.view.Menu
+import android.view.MenuItem
+import android.view.MotionEvent
+import android.view.SearchEvent
+import android.view.View
+import android.view.Window
+import android.view.WindowManager
+import android.view.accessibility.AccessibilityEvent
+import androidx.annotation.RequiresApi
+import androidx.core.view.GestureDetectorCompat
+import com.datadog.android.core.internal.utils.sdkLogger
+import java.lang.Exception
+
+internal class WindowCallbackWrapper(
+    val wrappedCallback: Window.Callback,
+    val gesturesDetector: GestureDetectorCompat
+) : Window.Callback {
```

Comment:
you can do 
```kotlin
internal class WindowCallbackWrapper(
    val wrappedCallback: Window.Callback,
    val gesturesDetector: GestureDetectorCompat
) : Window.Callback by wrappedCallback {
  // …
}
```
and only override the `dispatchTouchEvent` method (the rest is generated automatically by Kotlin :) )

Reply:
Oh man....and I don't have to struggle to write all these unit tests ...oh man :))